### ISSUE TITLE
Error when creating the ValidationError in some cases

### ISSUE THREAD

**Issue by SBRK** [Automattic/mongoose#5366]
In some cases, I'm getting an error when mongoose creates a `ValidationError`, when creating the `MongooseError` object.
It only happens in production and staging environments, and not on our development and local environments, even though we're using the same Node version.
It is a problem because it crashes the app at places where the `ValidationError` is seemingly created but not thrown yet (ex: setting an invalid value on a document, where the error is only thrown once calling save. Here, it crashes when setting the value).

```
TypeError: Cannot set property message of [object Object] which has only a getter
    at new MongooseError (/node/myapp/node_modules/mongoose/lib/error.js:16:3)
    at MongooseError.ValidationError (/node/myapp/node_modules/mongoose/lib/error/validation.js:20:19)
    at model.Document.invalidate (/node/myapp/node_modules/mongoose/lib/document.js:1592:32)
```

I'm not sure yet of the core reason, since I'm using the same version of Node both in local and in production. I wonder if anyone ran in that problem before.

Maybe changing to this in `lib/error.js` would fix the problem ?

``` javascript
function MongooseError(msg) {
  this.message = msg; // Setting message before calling Error.call
  Error.call(this);
  if (Error.captureStackTrace) {
    Error.captureStackTrace(this);
  } else {
    this.stack = new Error().stack;
  }
  // REMOVING THIS LINE THAT CAUSES THE CRASH
  //this.message = msg; 
  //
  this.name = 'MongooseError';
}
```

**Comment by davebaol** [comment]
Ha-ha the `TypeError: Cannot set property message of [object Object] which has only a getter` strikes again 😉 
See issue https://github.com/Automattic/mongoose/issues/5359

**Comment by vkarpov15** [comment]
Yeah this looks like it's a duplicate of #5359. Will fix for next patch release.

**Comment by vkarpov15** [comment]
I believe this was fixed with #5359 in 4.10.8.
